### PR TITLE
mapfishapp - buffer service

### PR DESCRIPTION
**Pull request by vdorut**
see https://github.com/georchestra/georchestra/issues/1015
Add buffer + testsuite

### CODE REVIEW

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/buffer/BufferController.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+public class BufferController {
+
+	
+	@RequestMapping(value = "/buffer/{bufferValue}?/{srs}", method = RequestMethod.POST, produces = "application/json; charset=UTF-8")
+	@ResponseBody
+	public String computeBuffer(@PathVariable double bufferValue,@RequestBody String wktgeom,@PathVariable String srs) throws Exception {
+
+		WKTReader w = new WKTReader();
+
+		Geometry g = w.read(wktgeom);
```
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/buffer/BufferController.java]: TODO: exception handling (eg: incorrect WKT syntax)

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/buffer/BufferController.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+		g = g.buffer(bufferValue);
```
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/buffer/BufferController.java]: same here ?

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/buffer/BufferController.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		StringBuilder json = new StringBuilder();
```
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/buffer/BufferController.java]: TODO: use JSONObject (as seen with @fgravin)

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/buffer/BufferController.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import com.vividsolutions.jts.geom.Geometry;
+
+@Controller
+public class BufferController {
+
+	
+	@RequestMapping(value = "/buffer/{bufferValue}?/{srs}", method = RequestMethod.POST, produces = "application/json; charset=UTF-8")
```
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/buffer/BufferController.java]: Okay, let's remove the srs parameter (if unused).

File: mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/buffer/BufferController.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	@ResponseBody
+	public String computeBuffer(@PathVariable double bufferValue,@RequestBody String wktgeom,@PathVariable String srs) throws Exception {
```
Comment (fvanderbiest) [inline mapfishapp/src/main/java/org/georchestra/mapfishapp/ws/buffer/BufferController.java]: Is it possible to wrap longer lines (>80 chars) ? Thanks :-)

**Comment by fvanderbiest** [comment]
Thanks Vincent, please address the above remarks, then squash your commits into a single one.

**Comment by vdorut** [comment]
@pmauduit

**Comment by fvanderbiest** [comment]
Now this looks good, thanks Vincent !
If @pmauduit agrees, we can merge.

**Pull request merged**